src/coreclr/interpreter/eeinterp.cpp

```diff
@@ -95,33 +61,36 @@ CorJitResult CILInterp::compileMethod(ICorJitInfo*         compHnd,
         return CORJIT_SKIPPED;
     }
 
-    InterpMethod *pMethod = InterpGetInterpMethod(methodInfo->ftn);
-    if (!pMethod->compiled)
-    {
-        InterpCompiler compiler(compHnd, methodInfo);
-        compiler.CompileMethod(pMethod);
-    }
+    InterpCompiler compiler(compHnd, methodInfo);
+    InterpMethod *pMethod = compiler.CompileMethod();
+
+    int32_t IRCodeSize;
+    int32_t *pIRCode = compiler.GetCode(&IRCodeSize);
  
     // FIXME this shouldn't be here
     compHnd->setMethodAttribs(methodInfo->ftn, CORINFO_FLG_INTERPRETER);
 
+    uint32_t sizeOfCode = sizeof(void*) + IRCodeSize * sizeof(int32_t);
```

Comment:
A nit - sizeof(void*) -> sizeof(InterpMethod*) to make it clear?

Reply:
I actually wonder - would there be any downside if  we stored the `InterpMethod` here directly instead of storing a pointer to it and allocating it separately?

Reply:
As the implementation progresses the size of `InterpMethod` will likely become quite large, while for a lot of simple methods it will be unnecessary. I see some of the common contents of it being moved to the code header for faster access, while less common ones to be accessed through an optional `InterpMethod*` pointer in the header. For simplicity at the beginning I say we just keep this fixed pointer instead and adjust it later as the implementation progresses.

Reply:
Ok, makes sense.